src/NHibernate/Linq/Visitors/NhPartialEvaluatingExpressionVisitor.cs

```diff
@@ -1,15 +1,110 @@
 using System;
 using System.Linq;
 using System.Linq.Expressions;
+using NHibernate.Collection;
 using Remotion.Linq.Clauses.Expressions;
 using Remotion.Linq.Parsing;
-using Remotion.Linq.Parsing.ExpressionVisitors;
 using Remotion.Linq.Parsing.ExpressionVisitors.TreeEvaluation;
 
 namespace NHibernate.Linq.Visitors
 {
+	//Modified version of PartialEvaluatingExpressionVisitor from Relinq (https://github.com/re-motion/Relinq)
+	//Copyright (c) rubicon IT GmbH, www.rubicon.eu
```

Comment:
According to https://tldrlegal.com/license/apache-license-2.0-(apache-2.0) we must include the license. So, I think it shall be included on the top of the file.

Reply:
I think lines differing from original implementation should be signaled by comments, for easing seeing what is done differently in NHibernate.